backend/src/main/java/de/laura/project/service/TriviaService.java

```diff
@@ -92,10 +91,94 @@ public boolean checkAnswer(int questionID, String selectedAnswer) {
-        return pointsMongoDB.save(triviaPointSummary);
+    public TriviaPointSummary savePoints(String user, int category, String difficulty, int amount, int points) {
+
+        List<TriviaPointDetails> currentGamePointDetailsList = new ArrayList<>();
+        TriviaPointDetails currentGamePointDetails = new TriviaPointDetails(difficulty, amount, points);
+        currentGamePointDetailsList.add(currentGamePointDetails);
+
+        List<TriviaPointCategory> currentGamePointCategoryList = new ArrayList<>();
+        TriviaPointCategory currentGamePointCategory = new TriviaPointCategory(category, currentGamePointDetailsList);
+        currentGamePointCategoryList.add(currentGamePointCategory);
+
+
+        TriviaPointSummary triviaPointSummaryFromMongoDb = pointsMongoDB.findById(user).get();
+        List<TriviaPointCategory> triviaPointCategoryFromMongoDbList = triviaPointSummaryFromMongoDb.getTriviaPointCategory();
+
+        boolean categoryExists = false;
+
+        for (TriviaPointCategory triviaPointCategoryFromDb : triviaPointCategoryFromMongoDbList) {
+            if (triviaPointCategoryFromDb.getCategory() == category) {
+                categoryExists = true;
+                if (triviaPointCategoryFromDb.getTriviaPointDetails().stream()
+                        .filter(triviaPointDetail -> triviaPointDetail.getDifficulty().equals(difficulty)).findAny().isEmpty()) {
+                    List<TriviaPointDetails> triviaPointDetailsFromMongoDbList = triviaPointCategoryFromDb.getTriviaPointDetails();
+                    triviaPointDetailsFromMongoDbList.add(currentGamePointDetails);
+                    triviaPointCategoryFromDb.setTriviaPointDetails(triviaPointDetailsFromMongoDbList);
+
+                } else {
+                    List<TriviaPointDetails> triviaPointDetailsFromMongoDbList = triviaPointCategoryFromDb.getTriviaPointDetails();
+                    List<TriviaPointDetails> adjustedTriviaPointDetails = triviaPointDetailsFromMongoDbList
+                            .stream().map(oldTriviaPointDetails -> new TriviaPointDetails(
+                                    oldTriviaPointDetails.getDifficulty(),
+                                    (oldTriviaPointDetails.getAmount() + amount),
+                                    (oldTriviaPointDetails.getPoints() + points))).collect(Collectors.toList());
+                    triviaPointCategoryFromDb.setTriviaPointDetails(adjustedTriviaPointDetails);
+                }
+                List<TriviaPointCategory> finalTriviaPointCategory = new ArrayList<>();
+                finalTriviaPointCategory.add(triviaPointCategoryFromDb);
+                triviaPointSummaryFromMongoDb.setTriviaPointCategory(finalTriviaPointCategory);
+            }
+        }
+        if (!categoryExists) {
+            triviaPointSummaryFromMongoDb.getTriviaPointCategory().add(currentGamePointCategory);
+        }
+        return pointsMongoDB.save(triviaPointSummaryFromMongoDb);
```

Comment:
Dennis teached me that one method should not be longer than 10 rows.

Reply:
Thanks!